### ISSUE TITLE
Нет возможности создания VM c gpu

### ISSUE THREAD

**Issue by Zikaf** [yandex-cloud/docker-machine-driver-yandex#28]
Здравствуйте, будет ли добавлена возможность создания vm с gpu?

**Comment by cheretbe** [comment]
Да, очень не хватает параметра аналогичному `--gpus` в `yc compute instance create`

**Comment by pitart** [comment]
gpu создает успешно, но пока недостаточно потестили, не могу гарантировать что не отвалилось создание виртуалок БЕЗ gpu. допилю еще https://github.com/yandex-cloud/docker-machine-driver-yandex/pull/29

**Comment by cheretbe** [comment]
ok, чтобы работало без gpu нам точно важнее)
Если нужна помощь в тестировании - могу посоздавать машины в тестовой среде

**Comment by Zikaf** [comment]
@pitart спасибо, очень ждем
Хотел уточнить так-же, будет ли решена проблема
команда:  docker-machine ls
"nable to query docker version: 400 Bad Request: {"message":"client version 1.15 is too old. Minimum supported API version is 1.24, please upgrade your client to a newer version"}"

По-моему на работу это не повлияло, сама проблема произошла когда с двух сторон были обновлены docker
Я правильно понял что требуется обновление в драйвере или проблема иная?

**Comment by pitart** [comment]
Тут у нас пока другая проблема - виртуалка то создается успешно, но если брать стандартный образ ubuntu-2204-lts, то в нем нет драйверов nvidia. А если брать ubuntu-2204-lts-gpu, то на нем есть драйвера, cuda и заодно докер. Видимо из-за этого, когда docker-machine начинает устанавливать на него докер - все падает с ошибкой. Пока не решим эту проблему - поддержка создания gpu в драйвере Яндекса - довольно бесполезная фича.

Running pre-create checks...
(default) Check folder exists
(default) Check if the instance with name "default" already exists in folder
Creating machine...
(default) Generating SSH Key
(default) Prepare an instance metadata (user-data included)
(default) Creating instance...
(default) Use image with ID "fd8oc4qnq5kg274e0vbn" from folder ID "standard-images"
(default) Waiting for Instance with ID "epdvd0rule5uifu6tutj"
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with ubuntu(systemd)...
Installing Docker...
Copying certs to the local machine directory...
Copying certs to the remote machine...
Setting Docker configuration on the remote daemon...
Error creating machine: Error running provisioning: Unable to verify the Docker daemon is listening: Maximum number of retries (10) exceeded

**Comment by Zikaf** [comment]
@pitart в моей инфре заранее сконфигурирован образ который содержит docker
Ошибки никакой соответственно нет, из образа у нас ВМ поднимается
и с images и с docker

**Comment by pitart** [comment]
тогда может быть ты знаешь какой-то волшебный способ сказать докер-машине о том, что она не должна устанавливать докер в ВМ?

**Comment by cheretbe** [comment]
Звучит очевидно, но мне помогло обновление бинаря docker-machine до последней версии при ошибке "client version x is too old. Minimum supported API version is y, please upgrade your client to a newer version" - не помню конкретных версий. После обновления ошибка исчезла

**Comment by Zikaf** [comment]
@cheretbe каким образом можно обновить на послед. версию?
Последнее обновление docker-machine в 19ом году, эта версия и выдает ошибку
т.к. docker уже 26 версии на моих vm

**Comment by cheretbe** [comment]
@Zikaf Ну да, он вообще-то deprecated. Но gitlab продолжает его использовать, поэтому пока поддерживает
https://gitlab.com/gitlab-org/ci-cd/docker-machine
https://gitlab.com/gitlab-org/ci-cd/docker-machine/-/releases/v0.16.2-gitlab.25/downloads/docker-machine-Linux-x86_64

**Comment by Zikaf** [comment]
@cheretbe этот релиз и использую, увы не помогает

**Comment by Zikaf** [comment]
@cheretbe А нет, прошу прощения
Внимательно поглядел
https://gitlab-docker-machine-downloads.s3.amazonaws.com/main/index.html
тут ласт версию взял, ошибки больше нет, спасибо

**Comment by pitart** [comment]
Если не смотреть на табуляцию - мой мр на добавление поддержки GPU готов.

**Comment by Zikaf** [comment]
@pitart спасибо большое
не уверен когда именно, но в обозримом будущем воспользуемся (сейчас в приоритете другие проекты)
если будут какие-то вопросы (ошибки), дам знать

**Comment by cheretbe** [comment]
@pitart А есть какая-то инструкция как собрать исполняемый файл?
В readme есть инструкция как запустить в среде go, но мне для тестов это не подойдет
